applications/FuzzLand.md

```diff
@@ -0,0 +1,366 @@
+
+For instance, consider following simple program:
+
+```rust
+// input: Vec<u8>
+if (input[0] > 20) { // Line 1
+    bug(); // Line 2
+} // Line 3
+```
+
+The CFG would be
+```
+   
+              ┌──────────────┐
+       ┌──────┤    Line 1    │
+       │      └───────┬──────┘
+       │ E2           │ E1
+       │      ┌───────▼──────┐
+       │   ┌──┤    Line 2    |
+       │   │  └──────────────┘          
+       │   │ E3              
+┌──────▼───▼───┐ 
+│    Line 3    |
+└──────────────┘ 
+```
+
+Given `u8` domain is 256, weight (exploration difficulty) of E1 is `(256 - 20) / (256 + 20)` and E2 is `(256 + 20) / (256 - 20)`. By intuition, E2 is indeed more likely to be explored than E1. As there is no comparison instruction in during transition of E3, the exploration difficulty is 0, meaning as long as we can reach Line 2, we can reach Line 3. 
+
+**Dynamic Program Analysis (DPA)**
+
+We support automated auditing of any program that can be compiled to LLVM bytecode by leveraging fuzz testing techniques, which involve sending random input to the program. This method, also known as heuristic search, aims to achieve 100% code coverage and uncover all vulnerabilities. While infinite time would guarantee zero false negatives, we use formal methods such as symbolic and concolic execution for guiding the fuzz testing search to reduce the time needed. Additionally, by partitioning the program into smaller, more manageable subprograms for each node, we can reduce the time required linearly as the number of nodes increases.
+
+Fuzz testing employs partitioning through the use of an instrumented target. If an input causes execution of code outside the partition plan, the target will terminate. Early termination reduces the time spent exploring code not within the partition, saving significant time.
+
+
+**Reaching Consensus**
+
+Verifying partition plans and interesting test cases can be costly or even impossible on the chain. Thus, validator nodes use off-chain oracles. FuzzLand uses rollup techniques to move the oracle results onto the chain and reach consensus. Specifically, an optimistic rollup pallet is implemented to achieve consensus on partition plans and interesting test cases. Once a validator node mints a partition plan or an auditor node mints a test case NFT, other validator nodes can submit fraud proofs to challenge it within 50 blocks, or it will be committed. Unlike human auditors or judges, validator nodes can find evidence to challenge false claims in microseconds, as the verification process is automated and inexpensive, making optimistic rollups effective. 
+
```

Comment:
interesting application scenario for roll-ups. worthwhile to explore ideas of mechanism design to ensure faster partition plan verification, maybe in a follow-up grant. ideas like [fair-cake-cutting](https://en.wikipedia.org/wiki/Fair_cake-cutting), allowing more decisions to be made on-chain

Reply:
Sounds good!